go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -59,17 +34,29 @@ type NodeController struct {
 	localPodLister listers.PodLister
 }
 
-// newNodeController returns a node handler that listens for node events
+// newNodeController returns a new node controller that runs on linux.
+// On linux there's two controller implementations. One runs from
+// ovnkube-node binary and the other from HO node binary and they
+// prepare the OVN and SDN nodes for the HO tunnel respectively.
 // so that Add/Update/Delete events are appropriately handled.
-// It initializes the node it is currently running on. On Linux, this means:
+// It initializes the node it is currently running on.
+// On ovn nodes, this means:
 //  1. Setting up a VXLAN gateway and hooking to the OVN gateway
 //  2. Setting back annotations about its VTEP and gateway MAC address to its own object
+//
+// On hybrid overlay nodes, this means:
+//  1. Add Hybrid Overlay DRMAC annotation to its own node object,
+//  2. Remove the ovnkube pod annotations from the pods running on this node.
 func newNodeController(
-	_ kube.Interface,
+	kube kube.Interface,
 	nodeName string,
 	nodeLister listers.NodeLister,
 	localPodLister listers.PodLister,
+	isHONode bool,
 ) (nodeController, error) {
+	if isHONode {
+		return newHONodeController(kube, nodeName, nodeLister, localPodLister)
+	}
 
 	node := &NodeController{
```

Comment:
nit: perhaps we could have a `newOVNNodeController` in `ovn_node_linux.go`

Reply:
fixed